### ISSUE TITLE
add functionality to toggle grouping monsters together

### ISSUE THREAD

**Issue by BrooklinJazz** [BrooklinJazz/dndBattleMaster#27]
# Goal
Toggle the grouping of monster combatants separate from the player combatants in CombatantList component. Allow the user to toggle the grouping of monsters for ease of play. This is so that DMs can keep monsters under the same initiative order if they prefer.
# Tasks
1. promote the toggling of grouping monsters from local state to a redux state value

2. if GroupMonsters is toggled (true), re-organize redux global value `CombatantList`  to have a grouped object amongst ungrouped objects (currently ungrouped objects are players and the grouped object is only monsters). this grouping should have it's own `InitiativeRoll` value for the purpose of sorting.

3. When GroupMonsters is toggled, the `monster_reducer.js` case 
ADD_MONSTER_TO_COMBATANTS should add to the grouped object within `CombatantList`. When it is not toggled it will perform the current default behavior of `Push`ing it to the end

4. Handle conditional rendering for `CombatantList.js` and `Combatant.js` that will group the monsters together when grouping is true.
    - add an organizer container component `InitiativesOrganizer.js` which sorts it's children by key `InitiativeRoll`
    - add a `CombatantGroup.js` component which contains and renders grouped `Combatant.js`s and sorts by their `InitiativeRoll`. This component should be re-usable given the potential for additional group types that may be added in the future. i.e. relics, structures, multi-level combat etc.

**Comment by cpb** [comment]
Don't forget about refactoring `Combatant` to be dumb. We were thinking of moving up the control logic into a wrapper which manages the initiative and removal and will render children.

I think the CombatantList-state will still be an array but I just want to clarify that for point 3 you're looking for the monsters group.

**Comment by BrooklinJazz** [comment]
For point 3 Combatant list state is always an array. There's an object within that array for grouped monsters.

After doing some work I realized that I may not need a wrapper due to the way I've organized the data. The object containing grouped monsters within Combatant list state has it's own InitiativeRoll value so a simple sort should work. Then I can move the logic for Combatant into CombatantList and refactor Combatant to a dummy component.

Then I'll make another dummy component for CombatantGroups.

My thinking is it's scallable and less heavy on performance.
